src/lib/currency/currency.ml

```diff
@@ -99,7 +105,7 @@ module Make_str (A : Wire_types.Concrete) = struct
 
     let precision_exp = Unsigned.of_int @@ Int.pow 10 precision
 
-    let to_formatted_string amount =
+    let string_of_mina_exn amount =
```

Comment:
This never fails, so `exn` should not be in the name.

Can we make this `to_mina_string`, and the complementary function `of_mina_string_exn`? 

A value of type `Currency.Balance.t` is a number in nanomina, but `mina_of_string_exn` suggests the result is something different than nanomina. If we think of the string as a "Mina string", that indicates that the string itself represents Mina.

Reply:
In fact, could we do that for the other functions, so that we'd have `of_mina_int_exn`, etc., instead of `mina_of_int_exn`? That way, it's clear that the `int` represents a number of Mina.

Reply:
Done in [4840454](https://github.com/MinaProtocol/mina/pull/11835/commits/4840454882cfa8c407593b6ed27f50eb50130595).